src/FixedPointDecimals.jl

```diff
@@ -38,6 +38,37 @@ import Base: reinterpret, zero, one, abs, sign, ==, <, <=, +, -, /, *, div, rem,
 
 include("fldmod_by_const.jl")
 
+# =========================================================
+# Set Int128 to widen to Int256 within FixedPointDecimals to avoid allocations from BigInt.
+# ---------------------------------------
+
+import BitIntegers
+
+# ---------- Necessary Extensions -----------------------
+# TODO: delete these once https://github.com/rfourquet/BitIntegers.jl/pull/2 and
+# https://github.com/rfourquet/BitIntegers.jl/pull/3 are merged.
+
+# BitIntegers is missing unsigned(::Type{Int256})
+# XBI == Union{BitIntegers.AbstractBitSigned, BitIntegers.AbstractBitUnsigned}
+Base.unsigned(::Type{T}) where T<:BitIntegers.XBI = typeof(convert(Unsigned, zero(T)))
+Base.unsigned(x::T) where T<:BitIntegers.XBI = reinterpret(unsigned(T), x)
+
+# BitIntegers is missing iseven/isodd
+# Prevent expensive calculation for Int256. Needed since `_round_to_even` calls `iseven`.
+Base.isodd(a::BitIntegers.Int256) = Base.isodd(a % Int)  # only depends on the final bit! :)
+# -------------------------------------------------------
+
+_widen(T::Type) = widen(T)
+_widen(x::T) where {T} = (_widen(T))(x)
+_widen(::Type{Int128}) = BitIntegers.Int256
+_widen(::Type{UInt128}) = BitIntegers.UInt256
+
+_widemul(a,b) = _widen(a)*_widen(b)
```

Comment:
Base's `widemul` actually has these two special cases:

```
widemul(x::Signed,y::Unsigned) = widen(x) * signed(widen(y))
widemul(x::Unsigned,y::Signed) = signed(widen(x)) * widen(y)
```

I believe we should replicate these here.

Reply:
huh weird, i wonder why those are needed...? But yeah okay sounds good to me!

Reply:
update: it turns out this is because apparently `widemul(::Signed, ::Unsigned)` returns a Signed, whereas promote would return Unsigned.

I've added your suggestion into the new take on this PR, here: https://github.com/JuliaMath/FixedPointDecimals.jl/pull/93